src/tracked_struct.rs

```diff
@@ -488,8 +491,9 @@ where
         // In that case we should not modify or touch it because there may be
         // `&`-references to its contents floating around.
         //
-        // Observing `Some(current_revision)` can happen in two scenarios: leaks (tsk tsk)
-        // but also the scenario embodied by the test test `test_run_5_then_20` in `specify_tracked_fn_in_rev_1_but_not_2.rs`:
+        // Observing `Some(current_revision)` can happen in two scenarios:
+        // - leaks, see tests\preverify-struct-with-leaked-data.rs and tests\preverify-struct-with-leaked-data-2.rs
+        // - or the following scenario (FIXME verify this? There is no test that covers this behavior):
```

Comment:
I am a bit curious about this second scenario. The test that used to exist here (its gone now) does not actually work anymore (I brought it back to check). So something within salsa's behavior must've changed here, @nikomatsakis any idea? Is this scenario working out differently nowadays, given you removed the test in one of your PRs? The comment was added in https://github.com/salsa-rs/salsa/commit/188f759555bed374435467f2df42a035dc834643#diff-545b847b1fec894075887ca68e52e020a61bc8f8e3d3ec6fe1bcbc27bb46b5d5 and the test was removed in https://github.com/nikomatsakis/salsa/commit/3d2b2d3a65288bfea50eba46cb63767d56fdf9a7